diff --git a/src/physics/cam/hetfrz_classnuc_cam.F90 b/src/physics/cam/hetfrz_classnuc_cam.F90
--- a/src/physics/cam/hetfrz_classnuc_cam.F90
+++ b/src/physics/cam/hetfrz_classnuc_cam.F90
@@ -28,10 +28,12 @@
 use cam_logfile,    only: iulog
 use error_messages, only: handle_errmsg, alloc_err
 use cam_abortutils, only: endrun
 
 use hetfrz_classnuc,   only: hetfrz_classnuc_init, hetfrz_classnuc_calc
+use micro_dv_nml, only: hclas_du_imm, hclas_du_dep, hclas_du_cnt, hclas_bc_imm, hclas_bc_dep, hclas_bc_cnt
+
 
 implicit none
 private
 save
 
@@ -802,13 +804,13 @@
                coated_aer_num(i,k,:), uncoated_aer_num(i,k,:), total_interstitial_aer_num(i,k,:), &
                total_cloudborne_aer_num(i,k,:), errstring)
 
             call handle_errmsg(errstring, subname="hetfrz_classnuc_calc")
 
-            frzimm(i,k) = frzbcimm(i,k) + frzduimm(i,k)
-            frzcnt(i,k) = frzbccnt(i,k) + frzducnt(i,k)
-            frzdep(i,k) = frzbcdep(i,k) + frzdudep(i,k)
+            frzimm(i,k) = hclas_bc_imm*frzbcimm(i,k) + hclas_du_imm*frzduimm(i,k)
+            frzcnt(i,k) = hclas_bc_cnt*frzbccnt(i,k) + hclas_du_cnt*frzducnt(i,k)
+            frzdep(i,k) = hclas_bc_dep*frzbcdep(i,k) + hclas_du_dep*frzdudep(i,k)
 
             if (frzimm(i,k) > 0._r8) freqimm(i,k) = 1._r8
             if (frzcnt(i,k) > 0._r8) freqcnt(i,k) = 1._r8
             if (frzdep(i,k) > 0._r8) freqdep(i,k) = 1._r8
             if ((frzimm(i,k) + frzcnt(i,k) + frzdep(i,k)) > 0._r8) freqmix(i,k) = 1._r8
