diff --git a/src/physics/cam/macrop_driver.F90 b/src/physics/cam/macrop_driver.F90
--- a/src/physics/cam/macrop_driver.F90
+++ b/src/physics/cam/macrop_driver.F90
@@ -21,10 +21,11 @@
   use cldwat2m_macro,    only: ini_macro
   use perf_mod,          only: t_startf, t_stopf
   use cam_logfile,       only: iulog
   use cam_abortutils,    only: endrun
   use zm_conv_intr,      only: zmconv_microp
+   use micro_dv_nml, only: detrainment_ramp_liq
 
   implicit none
   private
   save
 
@@ -721,11 +722,15 @@
       if( state_loc%t(i,k) > 268.15_r8 ) then
           dum1 = 0.0_r8
       elseif( state_loc%t(i,k) < 238.15_r8 ) then
           dum1 = 1.0_r8
       else
-          dum1 = ( 268.15_r8 - state_loc%t(i,k) ) / 30._r8
+          if( detrainment_ramp_liq ) then
+              dum1 = 0.0_r8
+          else
+              dum1 = ( 268.15_r8 - state_loc%t(i,k) ) / 30._r8
+          endif
       endif
 
      ! If detrainment was done elsewhere, still update the variables used for output
      ! assuming that the temperature split between liquid and ice is the same as assumed
      ! here.
