diff --git a/src/physics/cam/micro_mg_utils.F90 b/src/physics/cam/micro_mg_utils.F90
--- a/src/physics/cam/micro_mg_utils.F90
+++ b/src/physics/cam/micro_mg_utils.F90
@@ -33,13 +33,16 @@
 ! latvap          latent heat of vaporization                    J kg-1
 ! latice          latent heat of fusion                          J kg-1
 !
 !--------------------------------------------------------------------------
 
+use micro_dv_nml, only: onoff_contact_freezing, onoff_sip, rain_freeze, factor_HM
+
 #ifndef HAVE_GAMMA_INTRINSICS
 use shr_spfn_mod, only: gamma => shr_spfn_gamma
 #endif
+
 
 implicit none
 private
 save
 
@@ -1004,11 +1007,12 @@
 
   integer  :: i
 
   do i = 1,mgncol
 
-     if (qcic(i) >= qsmall .and. t(i) < 269.15_r8) then
+     !if (qcic(i) >= qsmall .and. t(i) < 269.15_r8) then
+     if (qcic(i) >= qsmall .and. t(i) < 269.15_r8 .and. onoff_contact_freezing) then
 
         if (.not. microp_uniform) then
            dum = var_coef(relvar(i), 4._r8/3._r8)
            dum1 = var_coef(relvar(i), 1._r8/3._r8)
         else
@@ -1166,14 +1170,16 @@
   real(r8), dimension(mgncol), intent(out) :: msacwi ! MMR
   real(r8), dimension(mgncol), intent(out) :: nsacwi ! Number
   integer :: i
 
   do i=1,mgncol
-     if((t(i) < 270.16_r8) .and. (t(i) >= 268.16_r8)) then
-        nsacwi(i) = 3.5e8_r8*(270.16_r8-t(i))/2.0_r8*psacws(i)
-     else if((t(i) < 268.16_r8) .and. (t(i) >= 265.16_r8)) then
-        nsacwi(i) = 3.5e8_r8*(t(i)-265.16_r8)/3.0_r8*psacws(i)
+     !if((t(i) < 270.16_r8) .and. (t(i) >= 268.16_r8)) then
+     if((t(i) < 270.16_r8) .and. (t(i) >= 268.16_r8) .and. onoff_sip ) then
+        nsacwi(i) = 3.5e8_r8*(270.16_r8-t(i))/2.0_r8*psacws(i)*factor_HM
+     !else if((t(i) < 268.16_r8) .and. (t(i) >= 265.16_r8)) then
+     else if((t(i) < 268.16_r8) .and. (t(i) >= 265.16_r8) .and. onoff_sip ) then
+        nsacwi(i) = 3.5e8_r8*(t(i)-265.16_r8)/3.0_r8*psacws(i)*factor_HM
      else
         nsacwi(i) = 0.0_r8
      endif
   enddo
 
@@ -1269,11 +1275,12 @@
   real(r8), dimension(mgncol), intent(out) :: nnuccr ! Number
   integer :: i
 
   do i=1,mgncol
 
-     if (t(i) < 269.15_r8 .and. qric(i) >= qsmall) then
+     !if (t(i) < 269.15_r8 .and. qric(i) >= qsmall) then
+     if (t(i) < 269.15_r8 .and. qric(i) >= qsmall .and. rain_freeze) then
         nnuccr(i) = pi*nric(i)*bimm* &
              (exp(aimm*(tmelt - t(i)))-1._r8)/lamr(i)**3
 
         mnuccr(i) = nnuccr(i) * 20._r8*pi*rhow/lamr(i)**3
 
